x-pack/platform/plugins/shared/entity_manager/server/lib/v2/definitions/get_identity_fields.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EntityV2 } from '@kbn/entities-schema';
+import { Logger } from '@kbn/core/server';
+import { readSourceDefinitions } from './source_definition';
+import { InternalClusterClient } from '../types';
+import { UnknownEntityType } from '../errors/unknown_entity_type';
+import { InvalidEntityInstance } from '../errors/invalid_entity_instance';
+
+export async function getIdentityFields(
+  type: EntityV2['entity.type'],
+  clusterClient: InternalClusterClient,
+  logger: Logger
+) {
+  const sources = await readSourceDefinitions(clusterClient, logger, {
+    type,
+  });
+
+  if (sources.length === 0) {
+    throw new UnknownEntityType(`No sources found for type ${type}`);
+  }
+
+  const identityFieldsBySource: { [key: string]: string[] } = {};
+
+  sources.forEach((source) => {
+    const { id, identity_fields: identityFields } = source;
+
+    identityFieldsBySource[id] = identityFields;
+  });
+
+  if (Object.keys(identityFieldsBySource).length === 0) {
+    throw new InvalidEntityInstance(
+      `Type ${type} is missing some identity fields, no sources could match`
+    );
```

Comment:
I would use a different error type and message here, since you're not looking at a specific instance here.

Also, I'm not sure if this case can even happen? This object would be empty if there were no sources to loop over which you already cover before, and we're not checking if `identityFields` is empty (which it shouldn't be due to other validation that happens during definition creation).

Reply:
removed that error as it was not providing any value